pkg/network/tracer/tracer.go

```diff
@@ -460,6 +460,16 @@ func (t *Tracer) GetActiveConnections(clientID string) (*network.Connections, er
 	return conns, nil
 }
 
+// ReleaseProtocolStats releases usm stats objects.
+func (t *Tracer) ReleaseProtocolStats(conn *network.Connections) {
+	t.bufferLock.Lock()
+	defer t.bufferLock.Unlock()
```

Comment:
why do you use this mutex?

Reply:
for precautionary reasons because it is used in Tracer.GetActiveConnections, however this lock here seems to be redundant. I'll remove it and check the impact.